webapp/app.py

```diff
@@ -73,11 +73,25 @@ def thank_you():
 
 @open_id.after_login
 def after_login(resp):
-    flask.session["macaroon_discharge"] = resp.extensions["macaroon"].discharge
-    flask.session["openid"] = {
-        "identity_url": resp.identity_url,
-        "email": resp.email,
+    root = flask.session["macaroon_root"]
```

Comment:
You don't absolutely have to do this, but it's good practice to add docstrings to all functions. See [PEP257](https://www.python.org/dev/peps/pep-0257/)